ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,306 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Third-party Imports
+from boto.ec2.elb.healthcheck import HealthCheck
+import boto.exception
+
+# Cloudify imports
+from ec2 import constants
+from ec2 import connection
+from ec2 import utils
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+
+@operation
+def creation_validation(**_):
+    """ This checks that all user supplied info is valid """
+
+    if not ctx.node.properties['resource_id']:
+        lb = None
+    else:
+        lb = _get_existing_load_balancer(
+            ctx.node.properties['resource_id'])
+
+    if ctx.node.properties['use_external_resource'] and not lb:
+        raise NonRecoverableError(
+            'External resource, but the supplied '
+            'elb does not exist in the account.')
+
+    if not ctx.node.properties['use_external_resource'] and lb:
+        raise NonRecoverableError(
+            'Not external resource, but the supplied '
+            'elb exists.')
+
+
+@operation
+def create_load_balancer(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        return None
+
+    lb = _create_load_balancer()
+
+    health_checks = ctx.node.properties.get('health_check')
+
+    if isinstance(health_checks, list):
```

Comment:
We should raise an error if isinstance is False, otherwise, why check in the first place? Personally, I don't think it's necessary to check, but I do like the idea of double checking. I would suggest that if you use it, you use it in all places to validate properties.